### PR TITLE
B 18429 ssw model to service object conversion

### PR DESCRIPTION
**Pull request by antgmann**
## [B-18429 PPM SSW Model to Service Object Conversion](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3a883383)

Integration PR: [B-18429-INT](https://github.com/transcom/mymove/pull/11735)

## Summary

The SSW was converted from a model to a service object, and as much data as possible was changed from personallyprocuredmove to PPMShipment. There are still obligations and signed certifications that will require downstream work, but this restructure was more time sensitive due to downstream dependencies.

Is there anything you would like reviewers to give additional scrutiny?

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Pull the branch, ensure the SSW files are correctly structured and no personallyprocuredmove code is still active.
2. Run the SSW Testing suite, ensure all tests pass and all tests have been reviewed.

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

### CODE REVIEW

**Review by r-mettler — APPROVED** [review]
Changes all match up with INT branch PR. 👍

File: pkg/handlers/internalapi/moves.go [hunk 1]
```diff
@@ -280,33 +279,37 @@ func (h SubmitMoveHandler) Handle(params moveop.SubmitMoveForApprovalParams) mid
...
+			ssfd.Obligations, err = h.SSWPPMComputer.ComputeObligations(appCtx, *ssfd, h.DTODPlanner())
 			if err != nil {
 				logger.Error("Error calculating obligations ", zap.Error(err))
 				return handlers.ResponseForError(logger, err), err
 			}
 
-			page1Data, page2Data, page3Data, err := models.FormatValuesShipmentSummaryWorksheet(ssfd)
+			page1Data, page2Data, page3Data := h.SSWPPMComputer.FormatValuesShipmentSummaryWorksheet(*ssfd)
 
 			if err != nil {
```
Comment (MInthavongsay) [inline pkg/handlers/internalapi/moves.go:311]: line 311 -- where is this err coming from? The diff on line 307 says err was removed from the original return implementation.

Comment (antgmann) [reply pkg/handlers/internalapi/moves.go:311]: This was an error that was improperly set up for when certain pieces of data were missing from the SSW. However, upon review and discussion with the team and customer, it was decided that the best way forward was not to treat these fields as required fields, hence why the error was removed.

Comment (MInthavongsay) [reply pkg/handlers/internalapi/moves.go:311]: Why then have the err nil check on line 311?

Comment (antgmann) [reply pkg/handlers/internalapi/moves.go:311]: So this ineffectual error call is not a show-stopping issue and is resolved in B-18428 ([see this PR](https://github.com/transcom/mymove/pull/11848)), which is the next downstream that this task enables, so it's not a pressing issue at the moment. Additionally, B-18061 removes this API endpoint entirely downstream.

**Review by MInthavongsay — APPROVED** [review]
OK approve. Anthony says it will resolve itself downstream.

**Pull request merged**